Add basic lemming node implementation

DanG100 opened the pull request.

alexmasi commented on topo/node/lemming/lemming.go:
```diff
@@ -0,0 +1,85 @@
+	if nodeImpl.Proto == nil {
+		return nil, fmt.Errorf("nodeImpl.Proto cannot be nil")
+	}
+	defaults(nodeImpl.Proto)
```
should use the return value here, even though it appears to modify the input in place as well

DanG100 replied:
done

alexmasi commented on topo/node/lemming/lemming.go:
```diff
@@ -0,0 +1,85 @@
+	*node.Impl
+}
+
+func (n *Node) ResetCfg(ctx context.Context) error {
```
PushConfig()
ResetCfg()
GenerateSelfSigned()

should all be implemented even if its just to return an "unimplemented" error with a message describing how to do the task (even if its as simple as config reset is not supported by Lemming, or config reset can be done by opening a cli on Lemming container and doing ...)

and you should include a block like: https://github.com/openconfig/kne/blob/f268e9f5c7adf28e4529dfa5aefa911124d9da23/topo/node/ceos/ceos.go#L51 to validate the signatures

alexmasi commented on topo/node/lemming/lemming.go:
```diff
@@ -0,0 +1,85 @@
+		pb.Config.EntryCommand = fmt.Sprintf("kubectl exec -it %s -- /bin/bash", pb.Name)
+	}
+	if pb.Services == nil {
+		pb.Services = map[uint32]*tpb.Service{
```
add ssh?

DanG100 replied:
ssh isn't enabled of the lemming container, leaving it off for now

alexmasi commented on topo/node/lemming/lemming.go:
```diff
@@ -0,0 +1,85 @@
+	return nil
+}
+
+func defaults(pb *tpb.Node) *tpb.Node {
```
add some labels and constraints? see other node implementations

DanG100 replied:
added

alexmasi commented on topo/node/lemming/lemming.go:
```diff
@@ -0,0 +1,114 @@
+}
+
+func (n *Node) ConfigPush(context.Context, io.Reader) error {
+	return fmt.Errorf("config push is not implemented using gNMI to configure device")
```
can you make these return status.Unimplemented grpc error codes?

DanG100 replied:
done

alexmasi commented on topo/node/lemming/lemming.go:
```diff
@@ -0,0 +1,114 @@
+)
+
+func (n *Node) ResetCfg(ctx context.Context) error {
+	return nil
```
maybe add a log here that resetcfg is a noop

DanG100 replied:
done

DanG100 commented:
@marcushines What do want to do about these lint issues? it would a somewhat significant breakage to fix

marcushines commented:
I would disable lint for those sections

DanG100 commented:
> I would disable lint for those sections and add comment to fix

Added symlink from .protolint.yaml into .github/linter to the github ci correctly used the config file. Added todo to remove the linter exclusions

alexmasi commented on .github/linters/.protolintrc.yml:
```diff
@@ -0,0 +1 @@
+../../.protolint.yaml
```
I think this file is from a time before the linter presubmit, can probably just rename it and move it here

DanG100 replied:
done
